src/openfl/display/BitmapData.hx

```diff
@@ -2105,36 +2105,63 @@ class BitmapData implements IBitmapDrawable {
 		
 	}
 	
-	
-	private function __renderCanvas (renderSession:RenderSession):Void {
+	#if (js && html5)
+	function __canBeDrawnToCanvas ():Bool {
 		
-		#if (js && html5)
-		if (!readable) return;
+		return image != null;
+		
+	} 
+	
+	
+	function __drawToCanvas (context:CanvasRenderingContext2D, transform:Matrix, roundPixels:Bool, pixelRatio:Float, scrollRect:Rectangle, useScrollRectCoords:Bool):Void {
 		
 		if (image.type == DATA) {
 			
 			ImageCanvasUtil.convertToCanvas (image);
 			
 		}
+
+		var scale = pixelRatio / this.__pixelRatio; // Bitmaps can have different pixelRatio than display, therefore we need to scale them properly
```

Comment:
In the past the `BitmapData.pixelRatio` was only taken into account for Bitmap, not for Tilemap. I think this worked, because we never had "runtime created" BitmapData (like CacheAsBitmap or Text) used as Tilemap. What I want to say, most likely it is right to handle it for Tilemap as well ;) What do you think?

Reply:
I'm not sure if it works correctly atm, but it definitely makes sense to handle pixel ratio for Tilemaps too, since we might want to have a Tilemap with hi-res atlas.